src/network/socks5.py

```diff
@@ -96,21 +111,25 @@ def state_pre_connect(self):
         return True
 
     def state_proxy_addr_1(self):
+        """"""
         self.boundaddr = self.read_buf[0:4]
         self.set_state("proxy_port", length=4, expectBytes=2)
         return True
 
     def state_proxy_addr_2_1(self):
+        """"""
         self.address_length = ord(self.read_buf[0:1])
         self.set_state("proxy_addr_2_2", length=1, expectBytes=self.address_length)
         return True
 
     def state_proxy_addr_2_2(self):
```

Comment:
Handle other addresses than IPv4 returned for peer (e.g. IPv6, onion, ...). This is part 2 which retrieves the data.